taker-frontend/src/useEventSource.ts

```diff
@@ -18,14 +18,14 @@ export function useEventSource(url: string): [EventSource | null, boolean] {
 
         let timer = setTimeout(() => {
             setIsConnected(false);
-        }, 5000);
+        }, 10000);
 
         const heartbeatHandler = () => {
             clearTimeout(timer);
             setIsConnected(true);
             timer = setTimeout(() => {
                 setIsConnected(false);
-            }, 5000);
+            }, 10000);
```

Comment:
🤔 We could add a `next_heartbeat` field to the heartbeat message and use that value + a buffer here. That way this value can never go out of sync with the backend.

Reply:
Nit: `next` would imply an absolute value? Just `interval` (or `pulse`? 😁) perhaps.